torch/_export/non_strict_utils.py

```diff
@@ -166,7 +166,7 @@ def make_fake_inputs(
             shape_env=ShapeEnv(
                 tracked_fakes=[],
                 co_fields=co_fields,
-                prefer_deferred_runtime_asserts_over_guards=allow_complex_guards_as_runtime_asserts,
+                prefer_deferred_runtime_asserts_over_guards=True,
```

Comment:
why keep this flag any more?

Reply:
The TLDR I think is that this flag helps avoid additional failures, but the asserts emitted are too messy to be on by default.

I guess this can be a longer discussion, but some guards still fail export unless this is specified; hybrid symints only handle the guards going through `SymNode.expect_true`. Guards that go through `guard_bool/guard_size_oblivious` won't get deferred, and these are typically complex guards from tracing lower-level op implementations. An example is in `test_export.py test_reshape_view_helper`:

```
class Model(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, x):
        return x.view(x.size(1), -1)
        # torch/_refs/__init__/_reshape_view_helper() will generate guards on reshape kernel(?)
        # Ne(s0, 20), so that reshape isn't no-op
        # Ne(Mod(s0, 20), 0), so that reshape needs to first flatten [s0, 20, 16] -> [s0*20, 16]
        # then split_dim -> [20, s0, 16]

model = Model()
x = torch.rand(1024, 20, 16)
dynamic_shapes = {"x": {0: Dim("batch")}}
```

If that flag is set, then export succeeds and these guards show up in the graph, but I think they seem too specialized to be on by default. There's some test cases that emit a ton more asserts with this flag, and I feel this is similar to exporting with real tensor prop, where we likely succeed but the asserts are a bit of a mess.